tests/integration/components/bs-modal-test.js

```diff
@@ -502,3 +502,44 @@ test('Renders in place (no wormhole) if renderInPlace is set', function(assert)
   assert.equal(this.$('.modal').length, 1, 'Modal exists.');
   assert.notEqual(this.$('.modal').parent().attr('id'), 'ember-bootstrap-modal-container');
 });
+
+test('Removes "modal-open" class when component is removed from view', function(assert) {
+  this.set('renderComponent', true);
+  this.render(hbs`{{#if renderComponent}}{{#bs-modal title="Simple Dialog"}}Hello world!{{/bs-modal}}{{/if}}<div id="ember-bootstrap-modal-container"></div>`);
+
+  let done = assert.async();
+
+  // wait for fade animation
+  setTimeout(() => {
+    assert.ok($('body').hasClass('modal-open'), 'body element has "modal-open" class.');
+
+    this.set('renderComponent', false);
+
+    // wait for fade animation
+    setTimeout(() => {
```

Comment:
No chance. Ember will remove the component from DOM immediately, there is nothing left to be animated! ;) That's why a model within an `{{#if ...}}` block won't have a chance to fade out, unless you trigger the fade out with setting `open` to false and set the if-condition to false in the `closed` action (that is called when the transition has finished) to fully remove it from DOM...